components/LeadFormDialog.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from "react";
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import MenuItem from "@mui/material/MenuItem";
+import { Lead, LeadStatus } from "@prisma/client";
+
+interface ErrorDisplayProps {
+  field: string;
+  errors?: { key: string; message: string }[];
+}
+
+function ErrorDisplay({ field, errors }: ErrorDisplayProps) {
+  return (
+    <>
+      {errors &&
+        errors.map((error) => {
+          if (error.key === field) {
+            return (
+              <div key={error.key} className="text-red-500">
+                {error.message}
+              </div>
+            );
+          }
+          return null;
+        })}
+    </>
+  );
+}
+
+interface LeadFormDialogProps {
+  leadData?: Lead;
+  onSubmit: (lead: Lead) => Promise<void>;
+  onClose: () => void;
+  errors?: { key: string; message: string }[];
+}
+
+export default function LeadFormDialog({
+  leadData,
+  onSubmit,
+  onClose,
+  errors,
+}: LeadFormDialogProps) {
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  const [lead, setLead] = React.useState<Lead>(
+    leadData ||
+      ({
+        name: "",
+        email: "",
+        status: "PROSPECT" as LeadStatus,
+        estimatedSaleAmount: 0,
+      } as Lead)
```

Comment:
Some type gymnastics here to make TS happy. Using the Omit keyword wasnt having the intended effect with the types generated by prisma client so I opted for this to keep working efficiently.